test/unit/api/tags.spec.ts

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable global-require */
+/* eslint-disable no-unused-expressions */
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import 'chai/register-should';
+import jwt from 'jsonwebtoken';
+import sleep from 'sleep-promise';
+import moment from 'moment';
+import { Errors as TagErrors } from 'server/routes/editTag';
+import app, { resetDatabase } from '../../../server-test';
+import { JWT_SECRET } from '../../../server/config';
+import * as modelUtils from '../../util/modelUtils';
+
+chai.use(chaiHttp);
+const { expect } = chai;
+
+describe('Tag Tests', () => {
+  const community = 'staking';
+  const chain = 'ethereum';
+
+  let adminJWT;
+  let adminAddress;
+  let userJWT;
+  let userAddress;
+  let tag;
+
+  before(async () => {
+    await resetDatabase();
+    let res = await modelUtils.createAndVerifyAddress({ chain });
+    adminAddress = res.address;
+    adminJWT = jwt.sign({ id: res.user_id, email: res.email }, JWT_SECRET);
+    const isAdmin = await modelUtils.assignRole({
+      address_id: res.address_id,
+      chainOrCommObj: { offchain_community_id: community },
+      role: 'admin',
+    });
+    expect(adminAddress).to.not.be.null;
+    expect(adminJWT).to.not.be.null;
+    expect(isAdmin).to.not.be.null;
+
+    res = await modelUtils.createAndVerifyAddress({ chain });
+    userAddress = res.address;
+    userJWT = jwt.sign({ id: res.user_id, email: res.email }, JWT_SECRET);
+    expect(userAddress).to.not.be.null;
+    expect(userJWT).to.not.be.null;
+  });
+
+  describe('/editTag', () => {
+    const name = 'test name';
+    const description = 'test description';
+    const featured_order = true;
+
+    before(async () => {
+      const markdownThread = require('../../util/fixtures/markdownThread');
+      const kind = 'forum';
+      const threadRes = await modelUtils.createThread({
+        chain,
+        address: userAddress,
+        jwt: userJWT,
+        title: decodeURIComponent(markdownThread.title),
+        body: decodeURIComponent(markdownThread.body),
+        privacy: true,
+        readOnly: true,
+        tags: ['tag', 'tag2', 'tag3'],
+        kind,
+      });
+      expect(threadRes.status).to.be.equal('Success');
+      expect(threadRes.result).to.not.be.null;
+      expect(threadRes.result.Address).to.not.be.null;
+      expect(threadRes.result.Address.address).to.equal(userAddress);
+      tag = threadRes.result.tags[0];
+    });
+
+    it('should successfully edit tag names & descriptions', async () => {
```

Comment:
if we're editing tag names and descriptions, I would put `newTagName` variable inside the test, not outside. Outside of tests is best for recurring data.

optionally, we can keep these variables outside the test, but indicate by their name its intended as a new name to be edited.